### PR TITLE
1909 object edit add icon section

### LINKED ISSUE #1909: Object Edit - Add Icon section
Warning: this ticket is part of a series of tickets aiming to create the "Object Edition flow". This is ticket number 3.

## Context

See #1765

For this issue:
- In the Object Edit page, add an "Icon" section with a title, a description, an icon picker and a preview. Use the `IconPicker` component.
- This section is shared with the New Object page (see #1808).

## Figma

Image

https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=11336%3A63490&mode=design&t=lrj7aulu2ylwpN5F-1

## Behavior

**GIVEN** I am on an Object Edit page
**THEN** I see an "Icon" section with a title, a description, an icon picker and a preview of the icon with the object name

**GIVEN** I am on an Object Edit page
**WHEN** I select an icon with the icon picker
**THEN** the preview displays the icon I picked
**AND** the object's icon is updated

### PR DESCRIPTION
**Pull request by bosiraphael**
- Added Icon Section
- Refactored IconPicker
- Updated its stories
- Created a useIconPicker hook

### CODE REVIEW

File: front/src/modules/settings/components/IconWithLabel.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  gap: ${({ theme }) => theme.spacing(1)};
+`;
+const StyledItemLabel = styled.div`
+  color: ${({ theme }) => theme.font.color.secondary};
+  font-size: ${({ theme }) => theme.font.size.sm};
+  font-style: normal;
+  font-weight: ${({ theme }) => theme.font.size.md};
+  leading-trim: both;
+  line-height: ${({ theme }) => theme.text.lineHeight.md};
+  text-edge: cap;
```
Comment (charlesBochet) [inline front/src/modules/settings/components/IconWithLabel.tsx:31]: introducing fancy css properties ;)

File: front/src/modules/settings/components/SettingsIconSection.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+import ArrowRight from '../assets/ArrowRight.svg';
+
+import { IconWithLabel } from './IconWithLabel';
+
+const StyledContainer = styled.div`
+  align-items: center;
+  align-self: stretch;
+  display: flex;
+  gap: 16px;
```
Comment (charlesBochet) [inline front/src/modules/settings/components/SettingsIconSection.tsx:14]: use theme for gap, margin, padding

File: front/src/modules/settings/components/SettingsIconSection.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import { IconPicker } from '@/ui/input/components/IconPicker';
+import { H2Title } from '@/ui/typography/components/H2Title';
+
+import ArrowRight from '../assets/ArrowRight.svg';
+
+import { IconWithLabel } from './IconWithLabel';
+
+const StyledContainer = styled.div`
+  align-items: center;
+  align-self: stretch;
```
Comment (charlesBochet) [inline front/src/modules/settings/components/SettingsIconSection.tsx:12]: is this align-self property useful? I see you add it to different containers but I am not sure it is needed

File: front/src/modules/settings/components/SettingsIconSection.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+const StyledArrowContainer = styled.div`
+  align-items: center;
+  display: flex;
+  height: 32px;
+  justify-content: center;
+`;
+
+type SettingsIconSectionProps = {
+  Icon: any;
```
Comment (charlesBochet) [inline front/src/modules/settings/components/SettingsIconSection.tsx:25]: this is not properly typed!

File: front/src/modules/ui/input/components/IconPicker.tsx [hunk 5]
```diff
@@ -63,28 +79,52 @@ export const IconPicker = ({ onChange, selectedIconKey }: IconPickerProps) => {
...
-          <DropdownMenuSkeletonItem />
-        ) : (
-          iconKeys.map((iconKey) => (
-            <StyledLightIconButton
-              aria-label={convertIconKeyToLabel(iconKey)}
-              isSelected={selectedIconKey === iconKey}
-              size="medium"
-              Icon={icons[iconKey]}
-              onClick={() => onChange({ iconKey, Icon: icons[iconKey] })}
+    <DropdownScope dropdownScopeId="icon-picker">
```
Comment (charlesBochet) [inline front/src/modules/ui/input/components/IconPicker.tsx:82]: nice!

File: front/src/modules/ui/input/components/__stories__/IconPicker.stories.tsx [hunk 6]
```diff
@@ -22,20 +22,79 @@ export const WithSelectedIcon: Story = {
   args: { selectedIconKey: 'IconCalendarEvent' },
 };
 
+export const WithOpen: Story = {
+  play: async ({ canvasElement }) => {
+    const canvas = within(canvasElement);
+
+    const inconPickerButton = await canvas.findByRole('button');
```
Comment (charlesBochet) [inline front/src/modules/ui/input/components/__stories__/IconPicker.stories.tsx:29]: typo incon

**Review by charlesBochet — COMMENTED** [review]
Nice work! Added a few comments!

File: front/src/modules/settings/components/IconWithLabel.tsx (hunk 1 above)
Comment (bosiraphael) [reply front/src/modules/settings/components/IconWithLabel.tsx:31]: It was on the figma file aha, should I let them in?

File: front/src/modules/settings/components/SettingsIconSection.tsx (hunk 3 above)
Comment (bosiraphael) [reply front/src/modules/settings/components/SettingsIconSection.tsx:12]: I will check

Comment (bosiraphael) [reply front/src/modules/settings/components/SettingsIconSection.tsx:12]: It doesn't seem useful here, I removed it in every file I added. It was on the figma css. This property is also used at several other locations in the code, we should maybe check if those are useful too.

Comment (charlesBochet) [reply front/src/modules/settings/components/SettingsIconSection.tsx]: Yes, I usually remove them when I refactor :p 
We should only introduce code purposefully!

**Review by charlesBochet — APPROVED** [review]
Approved :)

**Pull request merged**